open_rarity/rarity_ranker.py

```diff
@@ -53,18 +54,37 @@ def rank_collection(
         assert len(tokens) == len(scores)
 
         token_rarities: list[TokenRarity] = []
+
         # augment collection tokens with score information
         for idx, token in enumerate(tokens):
-            token_rarities.append(TokenRarity(token=token, score=scores[idx]))
+
+            # extract features from the token
+            token_features = (
+                TokenFeatureExtractor.extract_unique_attribute_count(
+                    token=token, collection=collection
+                )
+            )
+
+            token_rarities.append(
+                TokenRarity(
+                    token=token,
+                    score=scores[idx],
+                    token_features=token_features,
+                )
+            )
 
         return RarityRanker.set_rarity_ranks(token_rarities)
 
     @staticmethod
     def set_rarity_ranks(
         token_rarities: list[TokenRarity],
     ) -> list[TokenRarity]:
-        """Ranks a set of tokens Scores that are higher indicate a higher rarity,
-        and thus a lower rank.
+        """Ranks a set of tokens according to OpenRarity algorithm.
+        To account for unique items in collection OpenRarity implements
+        two-stage sorting ranking based on two factors:
```

Comment:
nit: double sorting? instead of two-stage

Reply:
makes sense will change